src/Symfony/Component/AssetMapper/ImportMap/ImportMapRenderer.php

```diff
@@ -21,34 +23,58 @@ class ImportMapRenderer
         private readonly string|false $polyfillUrl = ImportMapManager::POLYFILL_URL,
         private readonly array $scriptAttributes = [],
     ) {
     }
 
-    public function render(string $entryPoint = null, array $attributes = []): string
+    public function render(string|array $entryPointNames, array $attributes = []): string
     {
-        $attributeString = '';
+        $entryPointNames = (array) $entryPointNames;
+
+        $importMapData = $this->importMapManager->getImportMapData($entryPointNames);
+        $importMap = [];
+        $modulePreloads = [];
+        $cssLinks = [];
+        foreach ($importMapData as $importName => $data) {
+            $path = $data['path'];
+
+            if ($this->assetPackages) {
+                // ltrim so the subdirectory (if needed) can be prepended
+                $path = $this->assetPackages->getUrl(ltrim($path, '/'));
+            }
 
-        $attributes += $this->scriptAttributes;
-        if (isset($attributes['src']) || isset($attributes['type'])) {
-            throw new \InvalidArgumentException(sprintf('The "src" and "type" attributes are not allowed on the <script> tag rendered by "%s".', self::class));
+            $preload = $data['preload'] ?? false;
+            if ('css' === $data['type']) {
+                if ($preload) {
+                    $cssLinks[] = $path;
+                    // importmap entry is a noop
+                    $importMap[$importName] = 'data:application/javascript,';
+                } else {
+                    $importMap[$importName] = 'data:application/javascript,'.rawurlencode(sprintf('const d=document,l=d.createElement("link");l.rel="stylesheet",l.href="%s",(d.head||d.getElementsByTagName("head")[0]).appendChild(l)', $path));
+                }
+            } else {
+                $importMap[$importName] = $path;
+                if ($preload) {
+                    $modulePreloads[] = $path;
+                }
+            }
         }
 
-        foreach ($attributes as $name => $value) {
-            $attributeString .= ' ';
-            if (true === $value) {
-                $attributeString .= $name;
+        $output = '';
+        foreach ($cssLinks as $url) {
+            $url = $this->escapeAttributeValue($url);
 
-                continue;
-            }
-            $attributeString .= sprintf('%s="%s"', $name, $this->escapeAttributeValue($value));
+            $output .= "\n<link rel=\"stylesheet\" href=\"$url\">";
         }
 
-        $output = <<<HTML
-            <script type="importmap"{$attributeString}>
-            {$this->importMapManager->getImportMapJson()}
+        $scriptAttributes = $this->createAttributesString($attributes);
+        $importMapJson = json_encode(['imports' => $importMap], \JSON_THROW_ON_ERROR | \JSON_PRETTY_PRINT | \JSON_UNESCAPED_SLASHES | \JSON_HEX_TAG);
+        $output .= <<<HTML
+            <script type="importmap"{$scriptAttributes}>
+            {$importMapJson}
```

Comment:
let's be consistent (or use brackets everywhere):
```suggestion
            <script type="importmap"$scriptAttributes>
            $importMapJson
```